### PR TITLE
Fix disabling for_each in trigger config and allow modifying deploy config

### PR DESCRIPTION
**Pull request by ByJacob**
Small improvers for code

## what
- fix disable for_each in trigger_configuration
- allow modify aws_codedeploy_deployment_config

## why
- when set create_default_sns_topic to false terraform have error with dynamic configuration
- all changes in aws_codedeploy_deployment_config requires recreate, but cannot delete deployment_config when is assigned to deployment group
```
Plan: 1 to add, 2 to change, 1 to destroy.
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.app-deploy.aws_codedeploy_deployment_config.default[0]: Destroying... [id=ftg-preprod-magento2-m2-cron-nginx]
Releasing state lock. This may take a few moments...

Error: DeploymentConfigInUseException: Deployment configuration UUID is in use by Deployment Group UUID
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -258,7 +275,7 @@ resource "aws_codedeploy_deployment_group" "default" {
   }
 
   dynamic "trigger_configuration" {
-    for_each = local.sns_topic_arn == null ? [0] : [1]
+    for_each = length(try(local.sns_topic_arn, "")) <= 0 ? [] : [0]
```
Comment (nitrocode) [inline main.tf:278]: - Why is this changed to `[0]` instead of `[1]` ?
- Why change the `== null` ?

File: main.tf [hunk 2]
```diff
@@ -60,9 +61,21 @@ resource "aws_codedeploy_app" "default" {
...
+    compute_platform             = jsonencode(var.compute_platform)
+  }
+
+  byte_length = 2
+}
+
 resource "aws_codedeploy_deployment_config" "default" {
   count                  = local.count
-  deployment_config_name = module.this.id
+  deployment_config_name = "${module.this.id}${module.this.delimiter}${random_id.deployment_config_suffix.hex}"
```
Comment (nitrocode) [inline main.tf:78]: I believe we need a separate label for this where the `${random_id.deployment_config_suffix.hex}` can be added as an attribute.

File: main.tf (hunk 1 above)
Comment (ByJacob) [reply main.tf:278]: 1. Because earlier [0] it was in the first place :D. But in fact, the value [1] will be more telling.
2. Previously, var.sns_topic_arn was set to null and module.sns_topic.sns_topic.arn could return an empty string. While this is logically impossible, I wanted to set the same types for those in the variables from which the local.sns_topic_arn ultimately is computed.

File: main.tf (hunk 2 above)
Comment (ByJacob) [reply main.tf]: I prepare separate label for it

**Comment by ByJacob** [comment]
@nitrocode can we process this PR ?

**Review by nitrocode — DISMISSED** [review]

**Comment by nitrocode** [comment]
@ByJacob thanks for the contribution, working with me through the suggested changes, and being patient.

I've reached out to my team just to double check. if there aren't additional changes requested, we can merge

**Review by jamengual — DISMISSED** [review]
